casper/src/main/scala/coop/rchain/casper/rholang/InterpreterUtil.scala

```diff
@@ -83,6 +85,8 @@ object InterpreterUtil {
                      maxSeqNums = Map[Validator, Long](block.sender -> 0L)
                    ).pure[F]
                  }
+      rand = if (parentsSet.nonEmpty) BlockRandomSeed.fromBlock(block)
+      else BlockRandomSeed.fromGenesis(block)
```

Comment:
Why not do this check inside `BlockRandomSeed.fromBlock`?

Reply:
Genesis is different.